### PR TITLE
Fix new Creature Movement Bugs

### PR DESCRIPTION
**Pull request by Footman**
Co-authored-by: Francesco Borzì <borzifrancesco@gmail.com>

## Changes Proposed:
- fix UpdateEnvironmentIfNeeded was changing flying creature status all the time. closes: #4447
- fix hovering state was causing bugs at #4466
- add animation state priortiy: fly > hover > ground

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -2873,11 +2883,15 @@ bool Creature::SetHover(bool enable, bool packetOnly /*= false*/)
...
-        RemoveByteFlag(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_HOVER);
+   if (IsAlive() && !HasUnitState(UNIT_STATE_ROOT) && !HasUnitMovementFlag(MOVEMENTFLAG_ROOT))
+    {
+        if (IsLevitating())
+            SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_FLY);
+        else if (IsHovering())
+            SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_HOVER);
+        else
+            SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_GROUND);
+    }
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Creature/Creature.cpp:2894]: @Footman why don't just wrap this code into a method as it's repeated twice?

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @pklloveyou @mpfans please test this PR, it's related to the movement/pathfinding changes

**Comment by locus313** [comment]
tested, looks good.

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by disclosurez** [comment]
this issue is still happening on rimefang etc

https://i.imgur.com/0GOhoTH.jpg

**Comment by disclosurez** [comment]
lana'thel too

https://cdn.discordapp.com/attachments/813104786043699220/814572752617996348/unknown.png

**Pull request merged**